File: python/Ram.py
from Komponen import Komponen

class Ram(Komponen):
    def __init__(self, id="", warranty_months=0, merk="", nama="", harga=0.0,
                 kapasitas_gb=0, ddr="", clock_speed=0, power_consumption=0):
        super().__init__(id, warranty_months, merk, nama, harga)
        self.kapasitas_gb = kapasitas_gb
        self.ddr = ddr
        self.clock_speed = clock_speed
        self.power_consumption = power_consumption

    def set_kapasitas_gb(self, kapasitas_gb):
        self.kapasitas_gb = kapasitas_gb

    def set_ddr(self, ddr):
        self.ddr = ddr

    def set_clock_speed(self, clock_speed):
        self.clock_speed = clock_speed

    def set_power_consumption(self, watts):
        self.power_consumption = watts

    def get_kapasitas_gb(self):
        return self.kapasitas_gb

    def get_ddr(self):
        return self.ddr

    def get_clock_speed(self):
        return self.clock_speed

    def get_power_consumption(self):
        return self.power_consumption

    def display_info(self):
        super().display_info()
        print(f"RAM Specs: {self.kapasitas_gb}GB {self.ddr}, {self.clock_speed}MHz")
        print(f"Power Consumption: {self.power_consumption}W")

File: java/RAM.java
public class RAM extends Komponen {
    private int kapasitasGB;
    private String ddr;
    private int clockSpeed;
    private int powerConsumption;

    public RAM() {
        super();
        this.kapasitasGB = 0;
        this.ddr = "";
        this.clockSpeed = 0;
        this.powerConsumption = 0;
    }

    public RAM(String id, int warrantyMonths, String merk, String nama, double harga,
               int kapasitasGB, String ddr, int clockSpeed, int powerConsumption) {
        super(id, warrantyMonths, merk, nama, harga);
        this.kapasitasGB = kapasitasGB;
        this.ddr = ddr;
        this.clockSpeed = clockSpeed;
        this.powerConsumption = powerConsumption;
    }

    public void setKapasitasGB(int kapasitasGB) {
        this.kapasitasGB = kapasitasGB;
    }

    public void setDdr(String ddr) {
        this.ddr = ddr;
    }

    public void setClockSpeed(int clockSpeed) {
        this.clockSpeed = clockSpeed;
    }

    public void setPowerConsumption(int powerConsumption) {
        this.powerConsumption = powerConsumption;
    }

    public int getKapasitasGB() {
        return kapasitasGB;
    }

    public String getDdr() {
        return ddr;
    }

    public int getClockSpeed() {
        return clockSpeed;
    }

    public int getPowerConsumption() {
        return powerConsumption;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("RAM Specs: " + kapasitasGB + "GB " + ddr + ", " + clockSpeed + "MHz");
        System.out.println("Power Consumption: " + powerConsumption + "W");
    }
}
